ament_cppcheck/ament_cppcheck/main.py

```diff
@@ -229,28 +221,47 @@ def get_files(paths, extensions):
     return [os.path.normpath(f) for f in files]
 
 
-def get_xunit_content(report, testname, elapsed):
+def get_xunit_content(report, testname, elapsed, skip=None):
     test_count = sum(max(len(r), 1) for r in report.values())
     error_count = sum(len(r) for r in report.values())
     data = {
         'testname': testname,
         'test_count': test_count,
         'error_count': error_count,
         'time': '%.3f' % round(elapsed, 3),
+        'skip': test_count if skip else 0,
     }
     xml = """<?xml version="1.0" encoding="UTF-8"?>
 <testsuite
   name="%(testname)s"
   tests="%(test_count)d"
   failures="%(error_count)d"
   time="%(time)s"
+  skip="%(skip)d"
 >
 """ % data
 
     for filename in sorted(report.keys()):
         errors = report[filename]
 
-        if errors:
+        if skip:
+            data = {
+              'quoted_name': quoteattr(filename),
+              'testname': testname,
+              'quoted_message': quoteattr(''),
+              'skip': skip,
+            }
+            xml += """  <testcase
+    name=%(quoted_name)s
+    classname="%(testname)s"
+    status="notrun"
+  >
+      <skipped type="skip" message=%(quoted_message)s>
+        ![CDATA[Test Skipped due to %(skip)s]]
+      </skipped>
```

Comment:
NItpick: the indentation should be 2 spaces less - 2 spaces per level in the rest of the generated XML file.

Reply:
Fixed.